recipes/clang/activate-clang.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# This function takes no arguments
+# It tries to determine the name of this file in a programatic way.
+function _get_sourced_filename() {
+    if [ -n "${BASH_SOURCE[0]}" ]; then
+        basename "${BASH_SOURCE[0]}"
+    elif [ -n "${(%):-%x}" ]; then
+        # in zsh use prompt-style expansion to introspect the same information
+        # see http://stackoverflow.com/questions/9901210/bash-source0-equivalent-in-zsh
+        basename "${(%):-%x}"
+    else
+        echo "UNKNOWN FILE"
+    fi
+}
+
+# The arguments to this are:
+# 1. activation nature {activate|deactivate}
+# 2. toolchain nature {build|host|ccc}
+# 3. machine (should match -dumpmachine)
+# 4. prefix (including any final -)
+# 5+ program (or environment var comma value)
+# The format for 5+ is name{,,value}. If value is specified
+#  then name taken to be an environment variable, otherwise
+#  it is taken to be a program. In this case, which is used
+#  to find the full filename during activation. The original
+#  value is stored in environment variable CONDA_BACKUP_NAME
+#  For deactivation, the distinction is irrelevant as in all
+#  cases NAME simply gets reset to CONDA_BACKUP_NAME.  It is
+#  a fatal error if a program is identified but not present.
+function _tc_activation() {
```

Comment:
This function appears to be the same in both clang.sh and clang++.sh.  Can you move it into a third file which you source from these two, to prevent code duplication?

Reply:
If the anaconda recipe is fixed I'll change this in conda-forge, but till then, I'll keep this as it is so that it's easier for them to take our changes and for us to take their changes.

Reply:
Is this coming from an anaconda recipe? It looked like you wrote it :wink: (Also, I am on vacations now, so don't let me block this.)

Reply:
Yes, everything except the version bump came from the anaconda recipe. Anaconda recipe is for 4.0.1, this recipe copies it and bumps the version to 8.0.0